File: 3914-check-if-any-element-has-prime-frequency/3914-check-if-any-element-has-prime-frequency.cpp
class Solution {
private:
    bool isPrime(int n) {
        if (n <= 1)
            return false;
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0)
                return false;
        }
        return true;
    }
public:
    bool checkPrimeFrequency(vector<int>& nums) {
        int n = nums.size();
        array<int, 101> map;

        for (int i = 0; i < n; i++) {
            map[nums[i]]++;
        }

        for (int i = 0; i < 101; i++) {
            if (map[i] == 0)
                continue;
            if (isPrime(map[i]))
                return true;
        }
        return false;
    }
};

File: 3914-check-if-any-element-has-prime-frequency/3914-check-if-any-element-has-prime-frequency.py
class Solution:
    def checkPrimeFrequency(self, nums: List[int]) -> bool:
        n = len(nums)
        freq = [0] * 101

        def isPrime(num):
            if num <= 1:
                return False
            i = 2
            while i * i <= num:
                if num % i == 0:
                    return False
                i += 1

            return True
            
        for i in range(n):
            freq[nums[i]] += 1
            
        for i in range(101):
            
            if freq[i] == 0:
                continue
                
            if isPrime(freq[i]):
                return True
                
        return  False